pennylane/logging/configure.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains support methods for configuring the logging functionality.
+"""
+import logging
+import logging.config
+import os
+from importlib import import_module
+from importlib.util import find_spec
+
+has_toml = False
+toml_libs = ["tomllib", "tomli", "tomlkit"]
+for pkg in toml_libs:
+    spec = find_spec(pkg)
+    if spec:
+        tomllib = import_module(pkg)
+        has_toml = True
+        break
+
+# Define absolute path to this file in source tree
+_path = os.path.dirname(__file__)
```

Comment:
In my tests, I'm using a custom configuration file in a directory where I have my script and providing the full path. Works nicely!

Reply:
Sure, but that puts a limit on the path to require the file to be place in the same dir as the script, right? Can the file be elsewhere, like a system-wide configuration file that receives an absolute path?

Reply:
By providing the absolute path you can have it anywhere.

Reply:
But does this line not append the path to the config.py file directory to the passed string path?
```
with open(os.path.join(_path, config_file), "rb")
```
? I think I'll need to try this out, but I think that can cause issues.

Reply:
I tried it out. os.path.join will ignore the first element if the second is an absolute path. If the second is a relative path, it will keep it. In this case, if a user passes a relative path to their config file, this will fail. I think we need to identify how to handle all cases. The absolute path is the easiest thing to request, but I think we should also allow relative path support.

Reply:
[os.path.join](https://docs.python.org/3/library/os.path.html#os.path.join), is smart enough to analyze the paths provided in such a way that for:
```python
path1 = "/logging/pennylane/"
```
`os.path.join(path1, "collections")` will return `logging/pennylane/collections`
and 
`os.path.join(path1, "/collections")` will return `collections`
this way if we provide the absolute path for the second input, this will be what will come from it.

Reply:
Yes, I think one should be able to provide a relative (to the script) path.
Not clear how to do it, though.

Reply:
Some combination of https://docs.python.org/3/library/os.path.html#os.path.isabs and/or https://docs.python.org/3/library/os.path.html#os.path.isabs